torch/optim/adam.py

```diff
@@ -105,6 +105,8 @@ class Adam(Optimizer):
         capturable (bool, optional): whether this instance is safe to capture in a CUDA graph.
             Passing True can impair ungraphed performance, so if you don't intend to
             graph capture this instance, leave it False (default: False)
+        differentiable (bool,optional): use adaptative calculation of the derivative of 
```

Comment:
The differentiable kwarg here allows autograd to occur through the optimizer step in training; otherwise, the step() function runs in a torch.no_grad() context.

Could you also add a note similar to the one in capturable that passing True can impair performance, so if the user doesn't intend to run autograd through this instance to leave it False?

And don't forget to remove the trailing space 😛

Reply:
Yes, sure I will edit it :)

Reply:
is this more accurate?
use an adaptive calculation of the derivative, setting to True can impair the performance, setting to False disables running autograd through training instance as the step() function uses torch.no_grad()

Reply:
Hm, the adaptive calculation of the derivative is not entirely clear. I think it might be better to say:
```
differentiable (bool, optional): whether autograd should occur through the optimizer step in training; otherwise, the step() function runs in a torch.no_grad() context. Setting to True can impair performance, so leave it False if you don't intend to run autograd through this instance. (default: False) 
```

Reply:
Great, Thanks
I added it, 

> Hm, the adaptive calculation of the derivative is not entirely clear. I think it might be better to say:
> 
> ```
> differentiable (bool, optional): whether autograd should occur through the optimizer step in training; otherwise, the step() function runs in a torch.no_grad() context. Setting to True can impair performance, so leave it False if you don't intend to run autograd through this instance. (default: False) 
> ```

Thanks I added this, it is clearer